client/src/client.rs

```diff
@@ -430,6 +430,55 @@ impl Client {
         Ok(hash)
     }
 
+    /// Submit the prebuilt transaction ans wait until it is either rejected or committed.
+    /// If rejected, return the rejection reason.
+    ///
+    /// # Errors
+    /// Fails if sending a transaction to a peer fails or there is an error in the response
+    pub fn submit_transaction_blocking(
+        &self,
+        transaction: Transaction,
+    ) -> Result<HashOf<VersionedTransaction>> {
+        struct EventListenerInitialized;
+
+        let client = self.clone();
+        let (event_sender, event_receiver) = mpsc::channel();
+        let (init_sender, init_receiver) = mpsc::channel();
+        let hash = transaction.hash();
+        let _handle = thread::spawn(move || -> eyre::Result<()> {
+            let event_iterator = client
+                .listen_for_events(PipelineEventFilter::new().hash(hash.into()).into())
+                .wrap_err("Failed to establish event listener connection.")?;
+            init_sender
+                .send(EventListenerInitialized)
+                .wrap_err("Failed to send through init channel.")?;
```

Comment:
you need to specify what are you sending through the channel. the transaction? the init message?

Reply:
Init mesage, yes